cibuildwheel/windows.py

```diff
@@ -1,91 +1,132 @@
 from __future__ import print_function
 import os, tempfile, subprocess, shutil, sys
 from collections import namedtuple
+from contextlib import contextmanager
 from glob import glob
 
 try:
     from shlex import quote as shlex_quote
 except ImportError:
     from pipes import quote as shlex_quote
 
+try:
+    from urllib.request import urlopen
+except ImportError:
+    from urllib2 import urlopen
+
 from .util import prepare_command, get_build_verbosity_extra_flags
+if hasattr(sys, 'getwindowsversion'):
+    from ._vendored.pep514tools import findone as _pep514_findone
 
 
 IS_RUNNING_ON_AZURE = os.path.exists('C:\\hostedtoolcache')
 IS_RUNNING_ON_TRAVIS = os.environ.get('TRAVIS_OS_NAME') == 'windows'
+IS_RUNNING_ON_APPVEYOR = os.environ.get('APPVEYOR', 'false').lower() == 'true'
+
+
+@contextmanager
+def _mkdtemp():
+    path = tempfile.mkdtemp()
+    try:
+        yield path
+    finally:
+        shutil.rmtree(path)
```

Comment:
There's probably no need to clean up, as we're on a CI image. But maybe it's cleaner, indeed?

We should remember to streamline this on macOS and Linux as well, then, this cleaning up? (in some other commit/PR)

Reply:
It's cleaner indeed. Could be streamlined on other platforms indeed.